tests/src/testcommon.hh

```diff
@@ -255,42 +257,50 @@ template <Ikarus::ResultType resType>
   return t;
 }
 
-template <Ikarus::ResultType resType>
-[[nodiscard]] auto checkResultFunction(auto& nonLinearOperator, auto& fe, auto& feReq,
+template <Ikarus::ResultType resType, typename ResultEvaluator>
+[[nodiscard]] auto checkResultFunction(auto& nonLinearOperator, auto& fe, const auto& feRequirements,
+                                       const auto& expectedResult, const auto& evaluationPositions,
                                        const std::string& messageIfFailed = "") {
-  Dune::TestSuite t("Result Function Test");
-  using namespace Ikarus;
-  using namespace Dune::Indices;
-  using namespace Dune::Functions::BasisFactory;
+  Dune::TestSuite t("Result Function Test" + Dune::className(fe));
 
   using FiniteElement = std::remove_reference_t<decltype(fe)>;
+  auto& element       = fe.gridElement();
+  auto gridView       = fe.localView().globalBasis().gridView();
+  std::vector<FiniteElement> fes{fe};
+
+  Eigen::MatrixXd computedResults(expectedResult.rows(), expectedResult.cols());
+  auto localResultFunction =
+      Ikarus::ResultFunction<FiniteElement, resType, ResultEvaluator>::asLocalFunction(&fes, feRequirements);
```

Comment:
At call site your static functions you added to `ResultFunction` are quite verbose.

Can you move them out of class and make them as a free function? Then at call site you can let C++ deduce the template arguments and you can nicely write:

```cpp
auto localFunction = createlocalFunction<resType>(&fes, feRequirements);
```

Reply:
Or 

```cpp
auto localFunction = createlocalFunction<resType,ResultEvaluator>(&fes, feRequirements);
```

Reply:
See the comment above as more recent.